homeassistant/components/matrix/__init__.py

```diff
@@ -53,6 +55,9 @@
 CONF_WORD = "word"
 CONF_EXPRESSION = "expression"
 
+CONF_USERNAME_REGEX = "@[^:]*:.*"
+CONF_ROOMS_REGEX = "[!|#][^:]*:.*"
```

Comment:
```suggestion
CONF_ROOMS_REGEX = "^[!|#][^:]*:.*"
```

As they need to start with `!` or `#`